src/main/java/io/cdap/plugin/python/transform/PythonEvaluator.java

```diff
@@ -229,7 +230,7 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     }
     // try evaluating the script to fail application creation
     // if the script is invalid (only possible for interpreted mode)
-    if (config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) {
+    if (config.executionMode != null && config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) {
```

Comment:
please add a check here that script is not macro before we try to compile it

Reply:
```java
if (config.executionMode != null &&
    !config.containsMacro(Config.SCRIPT) &&
    config.executionMode.equalsIgnoreCase(EXECUTION_MODE_INTERPRETED)) 
```